### ISSUE TITLE
fix:Entire Page rebuilding mutiple times while onTap / focus on textfield

### ISSUE THREAD

**Issue by akshaynjarangal** [felangel/bloc#3330]
**Description**
Hey @felangel Entire Page rebuilding mutiple times while onTap / focus on textfield
**Steps To Reproduce**

1. Create a Stateless Widget
2. Adding bloc event. 
`WidgetsBinding.instance!.addPostFrameCallback((_) {context.read<DistrictListBloc>().add(LoadDistrictList()); // API DistrictList Calling Event
    });`

in build method
4. add ` log('something');`
5. Add TextFormField. 
6. Then ontap/focus on textfield entire widget rebuild
**Expected Behavior**
log output 

**Screenshots**
<img width="1440" alt="Screenshot 2022-04-19 at 1 14 19 PM" src="https://user-images.githubusercontent.com/63861163/163958167-d4c4f9b8-3d39-4d6f-8bd2-4b60b4a9b417.png">

**Comment by Add00w** [comment]
> 

could you provide minimal code that illustrates the issue?

**Comment by akshaynjarangal** [comment]
Hey @marcossevilla 👋 , 👉 [Dartpad link](https://dartpad.dev/?id=fdd369962f4ff6700a83c8a540fd6c4c)

**Comment by Add00w** [comment]
> Hey @marcossevilla 👋 , 👉 [Dartpad link](https://dartpad.dev/?id=fdd369962f4ff6700a83c8a540fd6c4c)

This is not showing your issue, what we need is a runnable code that reproduces your issue it doesn't matter _DartPad_ or _Github._

**Comment by akshaynjarangal** [comment]
This is Company production code, would you need my project GitHub repository access?

**Comment by Add00w** [comment]
> This is Company production code, would you need my project GitHub repository access?

Create a new flutter project and add the code with the issue then test if you have the issue upload to GitHub or Dartpad and share the link here.

**Comment by akshaynjarangal** [comment]
Hi @marcossevilla @Add00w  👋 [Flutter project on Github](https://github.com/akshaynjarangal/blocerrortest) , please test it.Hope you will find the error.

**Comment by Add00w** [comment]
> Hi @marcossevilla @Add00w 👋 [Flutter project on Github](https://github.com/akshaynjarangal/blocerrortest) , please test it.Hope you will find the error.

I cloned your repo, _signupload_  is printed first build, and when picking up date not when focusing on TextFields the only change I made is `- sdk: ">=2.16.2 <3.0.0"` to `+ sdk: ">=2.16.1 <3.0.0"`.

**Comment by Add00w** [comment]
> sdk version changed, but no change. @marcossevilla @Add00w check this video, when without keyboard hovering no error found, but when keyboard hovering error found, please use keyboard enable or use real device.
> 
>  Screen.Recording.mp4

[See this](https://user-images.githubusercontent.com/35359329/164411709-b1840f45-5a6b-42b7-83f1-5447f554ec0a.mov)

### I think you need 
`flutter clean`

**Comment by akshaynjarangal** [comment]
> > sdk version changed, but no change. @marcossevilla @Add00w check this video, when without keyboard hovering no error found, but when keyboard hovering error found, please use keyboard enable or use real device.
> > Screen.Recording.mp4
> 
>  Screen.Recording.2022-04-21.at.11.15.10.AM.mov 
> 
> ### I think you need
> `flutter clean`

After selection of date , then click on textfield, then issue will reproduce.please check

**Comment by Add00w** [comment]
> > > sdk version changed, but no change. @marcossevilla @Add00w check this video, when without keyboard hovering no error found, but when keyboard hovering error found, please use keyboard enable or use real device.
> > > Screen.Recording.mp4
> > 
> > 
> > Screen.Recording.2022-04-21.at.11.15.10.AM.mov
> > ### I think you need
> > `flutter clean`
> 
> After selection of date , then click on textfield, then issue will reproduce.please check and same issue with onchange function like dropdown ,date picking

Now I can see the issue but the DatePicker is causing the issue not the textField, since the showDatePicker is using the signup page context this issue exists , extract the date picker widget separately 
```
class DateOfBirthWidget extends StatelessWidget {
  const DateOfBirthWidget({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: () {
        //pick date
        context.read<PickDateBloc>().add(SelectDateofBirth(context: context));
      },
      child: BlocBuilder<PickDateBloc, PickDateState>(
        builder: (context, state) {
          return Container(
            width: MediaQuery.of(context).size.width,
            padding: const EdgeInsets.symmetric(horizontal: 25, vertical: 20),
            decoration: BoxDecoration(
              color: Colors.greenAccent,
              borderRadius: BorderRadius.circular(10),
            ),
            child: Text(
                state.dob == null
                    ? 'Date of Birth'
                    : DateFormat('dd MMMM yyyy').format(state.dob!),
                style: null),
          );
        },
      ),
    );
  }
}
```
or wrap it with the `Builder` widget.

**Comment by akshaynjarangal** [comment]
> > > > sdk version changed, but no change. @marcossevilla @Add00w check this video, when without keyboard hovering no error found, but when keyboard hovering error found, please use keyboard enable or use real device.
> > > > Screen.Recording.mp4
> > > 
> > > 
> > > Screen.Recording.2022-04-21.at.11.15.10.AM.mov
> > > ### I think you need
> > > `flutter clean`
> > 
> > 
> > After selection of date , then click on textfield, then issue will reproduce.please check and same issue with onchange function like dropdown ,date picking
> 
> Now I can see the issue but the DatePicker is causing the issue not the textField, since the showDatePicker is using the signup page context this issue exists , extract the date picker widget separately
> 
> ```
> class DateOfBirthWidget extends StatelessWidget {
>   const DateOfBirthWidget({Key? key}) : super(key: key);
> 
>   @override
>   Widget build(BuildContext context) {
>     return GestureDetector(
>       onTap: () {
>         //pick date
>         context.read<PickDateBloc>().add(SelectDateofBirth(context: context));
>       },
>       child: BlocBuilder<PickDateBloc, PickDateState>(
>         builder: (context, state) {
>           return Container(
>             width: MediaQuery.of(context).size.width,
>             padding: const EdgeInsets.symmetric(horizontal: 25, vertical: 20),
>             decoration: BoxDecoration(
>               color: Colors.greenAccent,
>               borderRadius: BorderRadius.circular(10),
>             ),
>             child: Text(
>                 state.dob == null
>                     ? 'Date of Birth'
>                     : DateFormat('dd MMMM yyyy').format(state.dob!),
>                 style: null),
>           );
>         },
>       ),
>     );
>   }
> }
> ```
> 
> or wrap it with the `Builder` widget.

Ok thank u bro  it's worked,🤗 let me check after API Connection , then I will close the issue .

**Comment by akshaynjarangal** [comment]
@Add00w when create new project and change sdk version , reproducing the issue

**Comment by akshaynjarangal** [comment]
@Add00w  @marcossevilla @felangel  sorry bro , finally cross check with flutter SDK, it's not error of flutter bloc , it's latest version flutter SDK bug 🥵

**Comment by excode** [comment]
Is the issue solved ?  

After upgrade to new Flutter 3 My app in android version start wried behaviour .

After through inspection I noticed that when I focus on any TextFieldEdit the on-Screen KeyBoard appear and App Goes one-Step Backward. It happen only in Android. IOS is working fine .
I have tested in Android Emulator as well as real device . As soon I focused on TextFieldEdit System-Keyboard appear and App goes 1 step backward.
